main/src/ca/uwaterloo/flix/api/lsp/Index.scala

```diff
@@ -180,6 +180,20 @@ case class Index(m: Map[(String, Int), List[Entity]],
     }
   }
 
+  /**
+    * Returns the expressions in the document at the given `uri` in the given range `range`.
+    */
+  def queryByRange(uri: String, range: Range): List[Entity] = {
+    (range.start.line to range.end.line).flatMap { line =>
+      m.getOrElse((uri, line), Nil)
+    }.filter { c =>
+      // Exclude all entities which are bigger than the given range.
+      c.loc.endLine <= range.end.line && c.loc.endCol <= range.end.character &&
```

Comment:
I think a helper function here would be useful.

Reply:
ok!